session.go

```diff
@@ -75,21 +81,22 @@ func (c *Client) login(ctx context.Context, password string) error {
 		PrivateKeyEncrypted string `xml:"privatekeyenc,attr"`
 	}
 	rsp := &response{}
-	if err = xml.NewDecoder(httpRsp.Body).Decode(rsp); err != nil {
-		return err
+	err = xml.NewDecoder(httpRsp.Body).Decode(rsp)
+	_ = httpRsp.Body.Close()
```

Comment:
I pushed a commit that checks the returned error.

Reply:
I don't think checking the error is required here. If the `Close` fails, there isn't a whole lot we can do about it. The reason I moved this out of `defer` (in addition to other similar `http.Response.Body.Close` calls) was to avoid dealing with defer inside of loops.

Since we are only interested in the `Body` for the purposes of `xml.Decoder.Decode` here, it seemed reasonable to just slam the door on the HTTP response and not other with the error.